modules/siddhi-core/src/main/java/io/siddhi/core/table/record/AbstractQueryableRecordTable.java

```diff
@@ -49,28 +70,525 @@
+                    compiledSelectionForCaching, outputAttributesForCaching);
+
+            if (preLoadedData != null) {
+                int preLoadedDataSize = findEventChunkSize(preLoadedData);
+                if (preLoadedDataSize <= maxCacheSize) {
+                    ComplexEventChunk<StreamEvent> loadedCache = new ComplexEventChunk<>();
+                    loadedCache.add(preLoadedData);
+                    cachedTable.addEvents(loadedCache, preLoadedDataSize);
+                } else {
+                    isCacheEnabled = false;
+                    log.warn(siddhiAppContext.getName() + ": " + cacheTableDefinition.getId() + " size is bigger " +
+                            "than cache table size defined as " + maxCacheSize + ". So cache is disabled");
+                }
+            }
+        }
+    }
+
+    private CompiledCondition generateCacheCompileCondition(Expression condition,
+                                                            MatchingMetaInfoHolder storeMatchingMetaInfoHolder,
+                                                            SiddhiQueryContext siddhiQueryContext,
+                                                            List<VariableExpressionExecutor>
+                                                                    storeVariableExpressionExecutors) {
+        MetaStateEvent metaStateEvent = new MetaStateEvent(storeMatchingMetaInfoHolder.getMetaStateEvent().
+                getMetaStreamEvents().length);
+        for (MetaStreamEvent referenceMetaStreamEvent: storeMatchingMetaInfoHolder.getMetaStateEvent().
+                getMetaStreamEvents()) {
+            metaStateEvent.addEvent(referenceMetaStreamEvent);
+        }
+        MatchingMetaInfoHolder matchingMetaInfoHolder = new MatchingMetaInfoHolder(
+                metaStateEvent,
+                storeMatchingMetaInfoHolder.getMatchingStreamEventIndex(),
+                storeMatchingMetaInfoHolder.getStoreEventIndex(),
+                storeMatchingMetaInfoHolder.getMatchingStreamDefinition(),
+                cacheTableDefinition,
+                storeMatchingMetaInfoHolder.getCurrentState());
+
+        Map<String, Table> tableMap = new ConcurrentHashMap<>();
+        tableMap.put(cacheTableDefinition.getId(), cachedTable);
+
+        return cachedTable.compileCondition(condition, matchingMetaInfoHolder,
+                storeVariableExpressionExecutors, tableMap, siddhiQueryContext);
+    }
+
+    protected abstract void connect() throws ConnectionUnavailableException;
 
     @Override
     public StreamEvent query(StateEvent matchingEvent, CompiledCondition compiledCondition,
                              CompiledSelection compiledSelection) throws ConnectionUnavailableException {
         return query(matchingEvent, compiledCondition, compiledSelection, null);
     }
 
+    public StreamEvent findInCache(CompiledCondition compiledCondition, StateEvent matchingEvent) {
```

Comment:
I think we don't need this method. As what is done here is primitive.

Reply:
it is used twice once in AQRT find and once in AQRT query

Reply:
its a one  line thing and when you use it inline you don't need to check for is cacheEnabled again